File: app/src/main/java/com/abs/launcher/LauncherApp.java
package com.abs.launcher;

import android.app.Application;

/**
 * Created by zy on 17-8-16.
 */

public class LauncherApp extends Application {

    private LauncherModel mLauncherModel;

    private static LauncherApp sInstance;

    public static LauncherApp getApp() {
        return sInstance;
    }

    @Override
    public void onCreate() {
        super.onCreate();
        sInstance = this;
    }

    public LauncherModel getModel() {
        return mLauncherModel;
    }
}

File: app/src/main/java/com/abs/launcher/LauncherApp.kt
package com.abs.launcher

import android.app.Application

/**
 * Created by zy on 17-12-18.
 */
class LauncherApp: Application() {

    companion object {
        lateinit var instance: LauncherApp
    }

    val model: LauncherModel by lazy { LauncherModel(this) }

    override fun onCreate() {
        super.onCreate()
        instance = this
    }
}